context.go

```diff
@@ -4,6 +4,7 @@ package tiledb
 #cgo LDFLAGS: -ltiledb
 #cgo linux LDFLAGS: -ldl
 #include <tiledb/tiledb.h>
+#include <tiledb/tiledb_experimental.h>
```

Comment:
> Well this API is experimental mostly because we don't know why allocs fail.

@Shelnutt2 were these the errors from REST reporting `Unhandled error in context allocation`? Just curious, I think those are fixed after [SC-56805](https://app.shortcut.com/tiledb-inc/story/56805/query-fails-with-unhandled-error-in-context-allocation-when-running-100-simultaneous-queries-on-rest-server) and wondering if we still want to merge this PR as-is to collect more info or https://github.com/TileDB-Inc/TileDB-Go/pull/225 instead? 

IMO if the errors are fixed and we don't expect to circle back in the near future going with #225 makes sense, it does separate the experimental and non-experimental APIs nicely.